liblwgeom/lwgeom.c

```diff
@@ -1549,6 +1549,21 @@ void lwgeom_set_srid(LWGEOM *geom, int32_t srid)
 
 /**************************************************************/
 
+static int
+point_cmp(const void *pn1, const void *pn2)
+{
+	LWPOINT *p1 = *((LWPOINT **)pn1);
+	LWPOINT *p2 = *((LWPOINT **)pn2);
+
+	GBOX b1, b2;
+	lwgeom_calculate_gbox((LWGEOM *)p1, &b1);
+	lwgeom_calculate_gbox((LWGEOM *)p2, &b2);
+
+	uint64_t h1, h2;
+	h1 = gbox_get_sortable_hash(&b1, 0);
```

Comment:
will probably be neater to refactor it to also allow points (it casts box back into a point internally and that piece can be isolated), and pass original srid